### DISCUSSION TITLE
automatically-validating-anti-forgery-tokens-in-asp-net-core-with-the-autovalidateantiforgerytokenattribute/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#215] [Announcements]
# automatically-validating-anti-forgery-tokens-in-asp-net-core-with-the-autovalidateantiforgerytokenattribute/

This post shows how to apply the AutoValidateAntiforgeryTokenAttribute globally to automatically protect all dangerous action methods from CSRF attacks.

https://andrewlock.net/automatically-validating-anti-forgery-tokens-in-asp-net-core-with-the-autovalidateantiforgerytokenattribute/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 21 2017, 07:51, in reply to [Anders Austad](https://disqus.com/by/anderaus/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/215#discussioncomment-6042649)</em>

---

<p>Great point, thanks!</p>

[a comment by an automated account is not shown]

**Comment by padhu-ar** [comment]
Hi andrew,
How can I log the exception using a custom logger if a missing antiforgery token occurs in a .NET Core application after adding the AutoValidateAntiforgeryToken globally in program.cs? Any suggestions would be greatly appreciated. Thank you in advance.
